src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java

```diff
@@ -80,7 +80,23 @@ public StudentExamResource(ExamAccessService examAccessService, StudentExamServi
     public ResponseEntity<StudentExam> getStudentExam(@PathVariable Long courseId, @PathVariable Long examId, @PathVariable Long studentExamId) {
         log.debug("REST request to get student exam : {}", studentExamId);
         Optional<ResponseEntity<StudentExam>> accessFailure = examAccessService.checkCourseAndExamAndStudentExamAccess(courseId, examId, studentExamId);
-        return accessFailure.orElseGet(() -> ResponseEntity.ok(studentExamService.findOneWithExercises(studentExamId)));
+
+        StudentExam studentExam = studentExamService.findOneWithExercises(studentExamId);
+
+        loadExercisesForStudentExam(studentExam);
+
+        // fetch participations, submissions and results for these exercises, note: exams only contain individual exercises for now
+        // fetching all participations at once is more effective
+        List<StudentParticipation> participations = participationService.findByStudentIdAndIndividualExercisesWithEagerSubmissionsResult(studentExam.getUser().getId(),
+                studentExam.getExercises());
+
+        // connect the exercises and student participations correctly and make sure all relevant associations are available
+        for (Exercise exercise : studentExam.getExercises()) {
+            // add participation with submission and result to each exercise
+            filterForExam(exercise, participations);
+        }
+
+        return accessFailure.orElseGet(() -> ResponseEntity.ok(studentExam));
```

Comment:
I think it would be better to return directly when an accessFailure occurs. This makes sense not only with regard to the procedure, but also because the access service checks whether the studentExam exists at all.

Reply:
fixed